### PR TITLE
[Component with its own class name as service type] LPS-198452 Refactor Wrapper component

### PR DESCRIPTION
**Pull request by Valmir-unicap**
Ticket: https://liferay.atlassian.net/browse/LPS-198452

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:startup

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/configuration/CPDefinitionLinkTypeSettingsImpl.java [hunk 1]
```diff
@@ -5,46 +5,40 @@
...
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
 
 /**
  * @author Alessio Antonio Rendina
  */
-@Component(service = CPDefinitionLinkTypeSettings.class)
+@Component(
+	configurationPid = "com.liferay.commerce.product.configuration.CPDefinitionLinkTypeConfiguration",
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/configuration/CPDefinitionLinkTypeSettingsImpl.java:22]: Hey @Valmir-unicap, please type on terminal `git grep com.liferay.commerce.product.configuration.CPDefinitionLinkTypeConfiguration` and you will see some usages of this configuration. So, this commit is breaking the current behavior.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
ci:stop

**Pull request closed without merging**